text/0233-opentelemetry-sandbox.md

```diff
@@ -0,0 +1,92 @@
+
+## Motivation
+
+Over the history of OpenTelemetry, there have been situations where people came to our community proposing interesting ideas to be adopted. There have also been vendors offering code donations to the project, some of which are now mostly unmaintained.
+
+As a possible solution to this, this OTEP proposes a new GitHub organization, [opentelemetry-sandbox](https://github.com/opentelemetry-sandbox). This organization will host projects until there's confidence that they have a healthy community behind them. They would also serve as a neutral place for the community to conduct experiments.
+
+The advantage of a sandbox organization is that OpenTelemetry's governance can be used, including Governance Committee (GC) and Technical Committee (TC) intermediation of conflicts, making sure it’s an inclusive place for people to collaborate while keeping the reputation of the OpenTelemetry project as a whole untouched, given that it would be clear that OpenTelemetry doesn’t officially support projects within the sandbox.
+
+There is a desire, but not an expectation, that projects will be moved from the sandbox as an official SIG or incorporated into an existing SIG. Realistically, we know that experiments might get dropped. There’s also no expectation that the OpenTelemetry project will provide resources to the sandbox project, like extra GitHub CI minutes or Zoom meeting rooms, although we might evaluate individual requests.
+
+This OTEP is inspired by [CNCF’s sandbox projects](https://www.cncf.io/sandbox-projects/), but the process is significantly different.
+
+## Internal details
+
+### Acceptance criteria
+
+A low barrier to entry would be desired for the sandbox. While the process can be refined based on our experience, the initial proposal for the process is the following:
+
+1. Proposals should be written following the template below and have one TC and/or GC sponsor, who will regularly provide the TC and GC information about the state of the project.
+2. Once a sponsor is found, the TC and GC will vote on accepting this new project on the Slack channel #opentelemetry-gc-tc.
+    1. After one week, the voting closes automatically, with the proposal being accepted if it has received at least one :thumbs-up: (that of the sponsor, presumably).
+    2. If at least one :thumbs-down: is given, or a TC/GC member has restrictions about the project but hasn’t given a :thumbs-down:, the voting continues until a majority is reached or the restrictions are cleared.
+    3. The voting closes automatically once a simple majority of the TC/GC electorate has chosen one side.
+3. Proponents should abide by OpenTelemetry’s Code of Conduct (currently the same as CNCF’s).
+4. There’s no expectation that small sandbox projects will have regular calls, but there is an expectation that all decisions will be made in public and transparently.
+5. Sandbox projects do NOT have the right to feature OpenTelemetry’s name on their websites.
```

Comment:
Can you explain the rational behind this? I am not against the statement, I am more worried about the "on their websites", why should a sandbox have a website at all?

Reply:
Perhaps this is me trying to anticipate the resolution of hypothetical problems, but I want to avoid something getting into the sandbox and then claiming all over the world that it is part of OTel, when it's just an experiment. A blessed experiment, but an experiment nonetheless.

Reply:
+1 for anticipating the hypothetical problem: my concern is an hypothetical as well: the "on their websites" bothers me, besides API/SDK docs I hope no SIG right now has their own website and everything stays either on github or opentelemetry.io 

> 5. Sandbox projects MUST communicate properly that their project is experimental in any circumstances
> 6. Sandbox projects MUST use resources and infrastructure (repository, slack channel, websites, etc.) provided by the OpenTelemetry community

Maybe that's to harsh, but this is what it boils down to for me.

Reply:
"on their websites" can be changed to "in their documentation". A github repo is a form of "website".

CNCF does not prohibit sandbox projects from using CNCF name, they just need to explicitly say "CNCF sandbox" (there are standard logos for those). I agree with @svrnm that "do NOT have the right" wording is odd and in contradiction with the stated benefits of the sandbox program.

Reply:
I like the requirement about being an explicit sandbox project. Perhaps this would be better?

```suggestion
5. Sandbox projects MAY mention being part of OpenTelemetry, and in doing so, they MUST explicitly state they are in the sandbox.
```

Reply:
I like that wording!